cookbooks/bcpc-hadoop/templates/default/generic_env.sh.erb

```diff
@@ -3,6 +3,7 @@
 # This file was generated by Chef.
 # Local changes may be reverted.
 #
+<% require 'shellwords' %>
 <% (@options || {}).each do |name, value| %>
-export <%= name.to_s %>=<%= value.to_s %>
+export <%= name.to_s %>=<%= value.to_s.shellescape %>
```

Comment:
This is a bad idea not everything needs to be shell escaped
````shell
env_sh[:YARN_CONF_DIR] = "${YARN_CONF_DIR:-$YARN_HOME/conf}"
```` becomes `export YARN_CONF_DIR=\$\{YARN_CONF_DIR:-\$YARN_HOME/conf\}`